webpack/assets/javascripts/react_app/components/common/TypeAheadSelect/TypeAheadSelectSelectors.js

```diff
@@ -0,0 +1,21 @@
+import Immutable from 'seamless-immutable';
+
+const selectTypeAheadSelect = ({ typeAheadSelect }, id) => typeAheadSelect[id];
+
+export const selectOptions = (state, id) => {
+  const typeAhead = selectTypeAheadSelect(state, id);
+  if (typeAhead && typeAhead.options) {
+    return Immutable.isImmutable(typeAhead.options)
+      ? typeAhead.options.asMutable()
+      : typeAhead.options;
+  }
+  return undefined;
+};
+
+export const selectSelected = (state, id) => {
+  const typeAhead = selectTypeAheadSelect(state, id);
+  if (typeAhead) {
+    return typeAhead.selected;
```

Comment:
If `selectTypeAheadSelect` will return in the end empty object `|| { }`
You could simpy do here something like: `return selectTypeAheadSelect(state, id).selected`